nuxeo-runtime/nuxeo-connect-standalone/src/main/java/org/nuxeo/launcher/connect/ConnectBroker.java

```diff
@@ -855,14 +878,50 @@ public boolean pkgInstall(List<String> packageIdsToInstall) {
      */
     public boolean pkgInstall(List<String> packageIdsToInstall, boolean ignoreMissing) {
         log.debug("Installing: " + packageIdsToInstall);
-        for (String pkgId : packageIdsToInstall) {
+        Queue<String> remaining = new LinkedList<>(packageIdsToInstall);
+        while (!remaining.isEmpty()) {
+            String pkgId = remaining.poll();
             if (pkgInstall(pkgId, ignoreMissing) == null && !ignoreMissing) {
                 return false;
             }
+            if (isRestartRequired()) {
```

Comment:
Why not store this in a variable before this Cycle to prevent String comparisons ?